internal/services/web/app_service_slot_custom_hostname_binding_resource.go

```diff
@@ -0,0 +1,193 @@
+
+	existing, err := client.GetHostNameBindingSlot(ctx, id.ResourceGroup, id.SiteName, id.SlotName, id.HostNameBindingName)
+	if err != nil {
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+		}
+	}
+
+	if existing.ID != nil && *existing.ID != "" {
+		return tf.ImportAsExistsError("azurerm_app_service_slot_custom_hostname_binding", *existing.ID)
+	}
+
+	properties := web.HostNameBinding{
+		HostNameBindingProperties: &web.HostNameBindingProperties{
+			SiteName: utils.String(id.SiteName),
+		},
+	}
+
+	if sslState != "" {
+		properties.HostNameBindingProperties.SslState = web.SslState(sslState)
+	}
+
+	if thumbprint != "" {
+		properties.HostNameBindingProperties.Thumbprint = utils.String(thumbprint)
+	}
+
+	if _, err := client.CreateOrUpdateHostNameBindingSlot(ctx, id.ResourceGroup, id.SiteName, id.HostNameBindingName, properties, id.SlotName); err != nil {
+		return fmt.Errorf("creating/updating %s: %+v", id, err)
+	}
+
+	d.SetId(id.ID())
+
+	return resourceAppServiceSlotCustomHostnameBindingRead(d, meta)
+}
+
+func resourceAppServiceSlotCustomHostnameBindingRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Web.AppServicesClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.AppServiceSlotCustomHostnameBindingID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.GetHostNameBindingSlot(ctx, id.ResourceGroup, id.SiteName, id.SlotName, id.HostNameBindingName)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[DEBUG] %s was not found - removing from state", id)
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("retrieving %s: %+v", *id, err)
+	}
+
+	slotId := parse.NewAppServiceSlotID(id.SubscriptionId, id.ResourceGroup, id.SiteName, id.SlotName)
+	d.Set("app_service_slot_id", slotId.ID())
+	d.Set("hostname", id.HostNameBindingName)
+
+	if props := resp.HostNameBindingProperties; props != nil {
+		d.Set("ssl_state", string(props.SslState))
+		d.Set("thumbprint", props.Thumbprint)
+		d.Set("virtual_ip", props.VirtualIP)
+	}
+
+	return nil
+}
+
+func resourceAppServiceSlotCustomHostnameBindingDelete(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Web.AppServicesClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.AppServiceSlotCustomHostnameBindingID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	locks.ByName(id.HostNameBindingName, appServiceSlotCustomHostnameBindingResourceName)
+	defer locks.UnlockByName(id.HostNameBindingName, appServiceSlotCustomHostnameBindingResourceName)
+
+	log.Printf("[DEBUG] deleting %s", id)
+
+	resp, err := client.DeleteHostNameBindingSlot(ctx, id.ResourceGroup, id.SiteName, id.SlotName, id.HostNameBindingName)
+	if err != nil {
+		if !utils.ResponseWasNotFound(resp) {
```

Comment:
just double-checking, is this necessary? the Delete function should be handling this

Reply:
copied this from the existing `app_service_custom_hostname_binding_resource.go`
https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/web/app_service_custom_hostname_binding_resource.go#L200